game-core/src/main/java/games/strategy/engine/framework/GameDataManager.java

```diff
@@ -79,14 +76,6 @@ public static GameData loadGame(final InputStream is) throws IOException {
       final Version readVersion = (Version) input.readObject();
       final boolean headless = HeadlessGameServer.headless();
       if (!GameEngineVersion.of(ClientContext.engineVersion()).isCompatibleWithEngineVersion(readVersion)) {
-        // a hack for now, but a headless server should not try to open any savegame that is not its version
-        if (headless) {
-          final String message = "Incompatible game save, we are: " + ClientContext.engineVersion()
-              + "  Trying to load game created with: " + readVersion;
```

Comment:
I unified the headed and headless messages here.  I didn't think it hurt to use the headed message in a headless context, as it simply includes the download link.

Reply:
I really hope one day it will just be interface implementations instead of checking if the client is headed or not ^^